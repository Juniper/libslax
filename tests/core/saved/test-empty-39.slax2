version 1.2;

param $first = 123123123123;
var $data := {
    <n> $first;
    <n> 56456456;
    <n> 789789;
    <n> 89789 + 512;
    <n> 987654321;
    <n> 8760000;
}

main <data> {
    for-each ($data/*) {
        <test n=.> {
            <out> slax:printf("%jH%12s", .);
            <out> slax:printf("%j{space}H%-12s", .);
            <out> slax:printf("%d + %jh%12s + %d", 128000, ., 256000);
            <out> slax:printf("%d + %j{space,whole,suffix=cool}h%-12s + %d", 128000, ., 256000);
        }
    }
    
    for-each (0 ... 3) {
        <out> slax:printf("%jt{love:}%12d", substring("abcdef", 1, . * 2));
        <out> slax:printf("%jt{more:}%12s", substring("abcdef", 1, . * 2));
    }
}
