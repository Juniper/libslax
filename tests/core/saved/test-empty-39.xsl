<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" xmlns:slax="http://xml.libslax.org/slax" version="1.0" extension-element-prefixes="slax-ext slax">
  <xsl:param name="first" select="123123123123"/>
  <xsl:variable name="data-temp-1">
    <n>
      <xsl:value-of select="$first"/>
    </n>
    <n>
      <xsl:value-of select="56456456"/>
    </n>
    <n>
      <xsl:value-of select="789789"/>
    </n>
    <n>
      <xsl:value-of select="89789 + 512"/>
    </n>
    <n>
      <xsl:value-of select="987654321"/>
    </n>
    <n>
      <xsl:value-of select="8760000"/>
    </n>
  </xsl:variable>
  <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="data" select="slax-ext:node-set($data-temp-1)"/>
  <xsl:template match="/">
    <data>
      <xsl:for-each select="$data/*">
        <test n="{.}">
          <out>
            <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:printf(&quot;%jH%12s&quot;, .)"/>
          </out>
          <out>
            <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:printf(&quot;%j{space}H%-12s&quot;, .)"/>
          </out>
          <out>
            <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:printf(&quot;%d + %jh%12s + %d&quot;, 128000, ., 256000)"/>
          </out>
          <out>
            <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:printf(&quot;%d + %j{space,whole,suffix=cool}h%-12s + %d&quot;, 128000, ., 256000)"/>
          </out>
        </test>
      </xsl:for-each>
      <xsl:for-each xmlns:slax="http://xml.libslax.org/slax" select="slax:build-sequence(0, 3)">
        <out>
          <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:printf(&quot;%jt{love:}%12d&quot;, substring(&quot;abcdef&quot;, 1, . * 2))"/>
        </out>
        <out>
          <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:printf(&quot;%jt{more:}%12s&quot;, substring(&quot;abcdef&quot;, 1, . * 2))"/>
        </out>
      </xsl:for-each>
    </data>
  </xsl:template>
</xsl:stylesheet>
