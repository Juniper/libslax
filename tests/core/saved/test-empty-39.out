<?xml version="1.0"?>
<data>
  <test n="1.23123123123e+11">
    <out>        123G</out>
    <out>123 G       </out>
    <out>128000 +         115G + 256000</out>
    <out>128000 + 115 Gcool    + 256000</out>
  </test>
  <test n="56456456">
    <out>         56M</out>
    <out>56 M        </out>
    <out>128000 +          54M + 256000</out>
    <out>128000 + 54 Mcool     + 256000</out>
  </test>
  <test n="789789">
    <out>        790K</out>
    <out>790 K       </out>
    <out>128000 +         771K + 256000</out>
    <out>128000 + 771 Kcool    + 256000</out>
  </test>
  <test n="90301">
    <out>         90K</out>
    <out>90 K        </out>
    <out>128000 +          88K + 256000</out>
    <out>128000 + 88 Kcool     + 256000</out>
  </test>
  <test n="987654321">
    <out>        988M</out>
    <out>988 M       </out>
    <out>128000 +         942M + 256000</out>
    <out>128000 + 942 Mcool    + 256000</out>
  </test>
  <test n="8760000">
    <out>        8.8M</out>
    <out>8.8 M       </out>
    <out>128000 +         8.4M + 256000</out>
    <out>128000 + 8 Mcool      + 256000</out>
  </test>
  <out>            </out>
  <out>            </out>
  <out>love:          ab</out>
  <out>more:          ab</out>
  <out>love:        abcd</out>
  <out>more:        abcd</out>
  <out>love:      abcdef</out>
  <out>more:      abcdef</out>
</data>
